File: Majo/practicas/calculadora.dart
import 'dart:io';

void main(){
  
  while (true){
    double num1 = obtenerNumero('Ingrese el primer número:');
    double num2 = obtenerNumero('Ingrese el segundo número:');
    String operacion = obtenerOperacion();

    double resultado = realizarOperacion(num1, num2, operacion);

    print('El resultado de $num1 $operacion $num2 es: $resultado');

    if (!deseaContinuar()) {
      break;
    }
  }
}
double obtenerNumero(String mensaje) {
  print(mensaje);
  return double.parse(stdin.readLineSync()!);
}

String obtenerOperacion() {
  print('Ingrese la operación (+, -, *, /):');
  return stdin.readLineSync()!;
}

double realizarOperacion(double num1, double num2, String operacion) {
  switch (operacion) {
    case '+':
      return sumar(num1, num2);
    case '-':
      return restar(num1, num2);
    case '*':
      return multiplicar(num1, num2);
    case '/':
      return dividir(num1, num2);
    default:
      print('Operación no válida');
      exit(1);
  }
}

double sumar(double a, double b) {
  return a + b;
}

double restar(double a, double b) {
  return a - b;
}

double multiplicar(double a, double b) {
  return a * b;
}

double dividir(double a, double b) {
  if (b == 0) {
    print('Error: No se puede dividir por cero');
    exit(1); 
  }
  return a / b;
}

bool deseaContinuar() {
  print('¿Desea realizar otra operación? (si/no)');
  String? respuesta = stdin.readLineSync();
  return respuesta?.toLowerCase() == 'si';
}

File: Majo/practicas/calculadora.rb
def main
    loop do
      num1 = obtener_numero('Ingrese el primer número:')
      num2 = obtener_numero('Ingrese el segundo número:')
      operacion = obtener_operacion
  
      resultado = realizar_operacion(num1, num2, operacion)
  
      puts "El resultado de #{num1} #{operacion} #{num2} es: #{resultado}"
  
      break unless desea_continuar
    end
  end
  
  def obtener_numero(mensaje)
    puts mensaje
    gets.chomp.to_f
  end
  
  def obtener_operacion
    puts 'Ingrese la operación (+, -, *, /):'
    gets.chomp
  end
  
  def realizar_operacion(num1, num2, operacion)
    case operacion
    when '+'
      sumar(num1, num2)
    when '-'
      restar(num1, num2)
    when '*'
      multiplicar(num1, num2)
    when '/'
      dividir(num1, num2)
    else
      puts 'Operación no válida'
      exit(1) 
    end
  end
  
  def sumar(a, b)
    a + b
  end
  
  def restar(a, b)
    a - b
  end
  
  def multiplicar(a, b)
    a * b
  end
  
  def dividir(a, b)
    if b == 0
      puts 'Error: No se puede dividir por cero'
      exit(1)  
    end
    a / b
  end
  
  def desea_continuar
    puts '¿Desea realizar otra operación? (si/no)'
    respuesta = gets.chomp.downcase
    respuesta == 'si'
  end
  
  main
  